front/pages/api/w/[wId]/vaults/index.ts

```diff
@@ -0,0 +1,141 @@
+import type { VaultType, WithAPIErrorResponse } from "@dust-tt/types";
+import { PostVaultRequestBodySchema, removeNulls } from "@dust-tt/types";
+import { isLeft } from "fp-ts/lib/Either";
+import * as reporter from "io-ts-reporters";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { withSessionAuthenticationForWorkspace } from "@app/lib/api/wrappers";
+import type { Authenticator } from "@app/lib/auth";
+import { GroupResource } from "@app/lib/resources/group_resource";
+import { UserResource } from "@app/lib/resources/user_resource";
+import { VaultResource } from "@app/lib/resources/vault_resource";
+import logger from "@app/logger/logger";
+import { apiError } from "@app/logger/withlogging";
+
+export type GetVaultsResponseBody = {
+  vaults: VaultType[];
+};
+
+export type PostVaultsResponseBody = {
+  vault: VaultType;
+};
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<
+    WithAPIErrorResponse<GetVaultsResponseBody | PostVaultsResponseBody>
+  >,
+  auth: Authenticator
+): Promise<void> {
+  const owner = auth.workspace();
+  if (!owner) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace you requested was not found.",
+      },
+    });
+  }
+
+  switch (req.method) {
+    case "GET":
+      const vaults = await VaultResource.listWorkspaceVaults(auth);
+
+      return res.status(200).json({
+        vaults: vaults
```

Comment:
I'm wondering if this should not be bundle within the VaultResource logic 🤔

Reply:
yes, it makes sense - let's do it a another pr